model-optimizer/mo/moc_frontend/extractor.py

```diff
@@ -1,35 +1,77 @@
 # Copyright (C) 2018-2021 Intel Corporation
 # SPDX-License-Identifier: Apache-2.0
 
-import logging as log
 import re
-from collections import defaultdict
-from copy import copy
-
-import numpy as np
 
+from mo.front.extractor import raise_no_node, raise_node_name_collision
 from mo.utils.error import Error
 
-from ngraph.frontend import InputModel  # pylint: disable=no-name-in-module,import-error
+from ngraph.frontend import InputModel, Place  # pylint: disable=no-name-in-module,import-error
+
+import numpy as np
 
 
 def decode_name_with_port(input_model: InputModel, node_name: str):
     """
     Decode name with optional port specification w/o traversing all the nodes in the graph
-    TODO: in future node_name can specify input/output port groups and indices (58562)
+    TODO: in future node_name can specify input/output port groups as well as indices (58562)
     :param input_model: Input Model
     :param node_name: user provided node name
     :return: decoded place in the graph
     """
-    # Check exact match with one of the names in the graph first
+    found_nodes = []
+    found_node_names = []
+
+    def add_found_node(found_nodes: list, node: Place):
+        """
+        Helper function to add Place node to list of found nodes. Adds only if node is not None
+        :param found_nodes List[Place]: list of found nodes
+        :param node Place: Place node to add to list
+        """
+        found_nodes.append(node) if node else None
+
     node = input_model.get_place_by_tensor_name(node_name)
-    if node:
-        return node
+    found_node_names.append('Tensor:' + node_name)
+    add_found_node(found_nodes, node)
+
+    node = input_model.get_place_by_operation_name(node_name)
+    add_found_node(found_nodes, node)
+    found_node_names.append('Operation:' + node_name)
+
+    regexp_post = r'(.+):(\d+)'
+    match_post = re.search(regexp_post, node_name)
+    if match_post:
+        found_node_names.append(match_post.group(1))
+        node_post = input_model.get_place_by_operation_name(match_post.group(1))
+        add_found_node(found_nodes, node_post.get_output_port(
+            outputPortIndex=int(match_post.group(2)))) if node_post else None
+
+    regexp_pre = r'(\d+):(.+)'
+    match_pre = re.search(regexp_pre, node_name)
+    if match_pre:
+        found_node_names.append(match_pre.group(2))
+        node_pre = input_model.get_place_by_operation_name(match_pre.group(2))
+        add_found_node(found_nodes, node_pre.get_input_port(
+            inputPortIndex=int(match_pre.group(1)))) if node_pre else None
+
+    if len(found_nodes) == 0:
+        raise_no_node(node_name)
+
+    # Check that there is no collision, all found places shall point to same data
+    if len(found_nodes) > 1:
+        all_equal = all([n.is_equal_data(found_nodes[0]) for n in found_nodes])
+        if not all_equal:
+            raise_node_name_collision(node_name, found_node_names)
```

Comment:
here we should print not all node_names in the list and print just names that have collision with `node[0]`

Reply:
This is 'decode_name_with_port'. So, `found_node_names` are actually represent all nodes matched pattern (like `0:1`).